python/CodeEditor.py

```diff
@@ -672,13 +672,15 @@ def RejectChange(index):
             content = line[2:]
             # restore deleted line
             log.debug(f"restore line {lineno}")
-            VimHelper.InsertLine(lineno, content)
+            VimHelper.InsertLine(lineno+1, content)
             lineno += 1
             g_restored_lines += 1
         elif (line.startswith('+ ')):
             # remove added line
             log.debug(f"delete line {lineno}")
-            VimHelper.DeleteLine(lineno)
+            VimHelper.DeleteLine(start_line)
+            g_restored_lines += 1
+            lineno += 1
```

Comment:
The current `lineno` does not change when deleting a line (undoing the add).
Maybe we need a `g_restored_lines -= 1` here? Incrementing makes no sense.